polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala

```diff
@@ -215,6 +228,58 @@ class PythonInterpreter private[python] (
     }
   }
 
+  private def definitionLocations(script: PyObject, line: Int, col: Int) = jep { jep =>
+    val pyNames = script.getAttr("goto", classOf[PyCallable]) // see https://jedi.readthedocs.io/en/latest/docs/api.html#jedi.Script.goto
+      .callAs(
+        classOf[Array[PyObject]],
+        Array[Object](Integer.valueOf(line), Integer.valueOf(col)), // positional args
+        Map[String, Object]("follow_imports" -> Boolean.box(true), "follow_builtin_imports" -> Boolean.box(true)).asJava)
+    pyNames.toList.flatMap {
+      jediName =>
+        val path = jediName.getAttr("module_path", classOf[String])
+        val line = jediName.getAttr("line", classOf[Integer])
+        val column = jediName.getAttr("column", classOf[Integer])
+        if (path != null)
+          Some(DefinitionLocation(s"python://$path", line, column))
+        else None
+    }
+  }
+
+  private def definitionSource(locations: List[DefinitionLocation]) = {
+    for {
+      firstLoc  <- ZIO(locations.headOption).someOrFailException
+      path      <- ZIO(Paths.get(firstLoc.uri.stripPrefix("python://")))
+      source    <- readFile(path)
+    } yield source
+  }.option
+
+  // TODO: we probably need to write out prior code cells, and use the jedi Project (https://jedi.readthedocs.io/en/latest/docs/api.html#jedi.Project)
+  //        to allow finding references to other cells. This will only find references that are external.
+  override def goToDefinition(code: String, pos: Int, state: State): RIO[Blocking with Logging, (List[DefinitionLocation], Option[String])] = {
+    for {
+      globals     <- populateGlobals(state)
+      (line, col)  = posToLineAndColumn(code, pos)
+      jedi        <- jep(_.getValue("jedi.Interpreter", classOf[PyCallable]).callAs(classOf[PyObject], code, Array(globals)))
```

Comment:
```suggestion
      script      <- jep(_.getValue("jedi.Interpreter", classOf[PyCallable]).callAs(classOf[PyObject], code, Array(globals)))
```

Reply:
It's a `jedi.Interpreter` though, not a `jedi.Script`. And it seems confusing to call it `interpreter`